rest/config_test.go

```diff
@@ -440,3 +441,369 @@ func TestAutoImportEnabled(t *testing.T) {
 		})
 	}
 }
+
+func TestMergeWith(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyAll)()
+	defaultInterface := "4984"
+	adminInterface := "127.0.0.1:4985"
+	profileInterface := "127.0.0.1:4985"
+	configServer := "remote.config.server:4985/db"
+	deploymentID := "DeploymentID1008"
+	facebookConfig := FacebookConfig{Register: true}
+
+	corsConfig := &CORSConfig{
+		Origin:      []string{"http://example.com", "*", "http://staging.example.com"},
+		LoginOrigin: []string{"http://example.com"},
+		Headers:     []string{},
+		MaxAge:      1728000,
+	}
+
+	deprecatedLog := []string{"Admin", "Access", "Auth", "Bucket", "Cache"}
+
+	databases := make(DbConfigMap, 2)
+	databases["db1"] = &DbConfig{Name: "db1"}
+	databases["db2"] = &DbConfig{Name: "db2"}
+	self := &ServerConfig{Databases: databases}
```

Comment:
Move this down (after 'other' is initialized', for clarity.

Reply:
This is fixed and latest changes are committed.